frame/pablo/src/lib.rs

```diff
@@ -981,6 +1012,79 @@ pub mod pallet {
 			lp_for_liquidity::<T>(pool, pool_account, base_amount, quote_amount)
 		}
 
+		fn update_accounts_deposited_one_asset_storage(
+			who: Self::AccountId,
+			pool_id: Self::PoolId,
+			lp_amount: Self::Balance,
+			action: SingleAssetAccountsStorageAction,
+		) -> Result<(), DispatchError> {
+			match action {
+				SingleAssetAccountsStorageAction::Depositing =>
+					if SingleAssetAccountsStorage::<T>::contains_key(&who, &pool_id) {
+						let lp_amount = SingleAssetAccountsStorage::<T>::try_mutate(
```

Comment:
you're doing 2 reads here which seems unnecessary

Reply:
@benluelo To be honest, I thought about it too but didn't come up with more exquisite version.

Reply:
From what I can tell, you should be able to move the `else` block into the `None` branch? You should also use `try_mutate` in the withdrawing action branch

Reply:
I understand your choice. But I'm afraid it's not entirely possible, you can't change the storage inside the `try_mutate` function. I mean, it's impossible to use `insert` function inside `try_mutate` function. 
But I [refactored](https://github.com/ComposableFi/composable/pull/1459/commits/4a1906b6b409ef23f68b69e8cb59a7bd3897ed89) the code a little bit, so now there is only one read of storage in each arm of match.

Reply:
Resolved!
Wow, you were right! Thanks a lot for this solution. Haven't used `try_mutate_exists` before.